solr/solrj/src/test/org/apache/solr/client/solrj/GetByIdTest.java

```diff
@@ -30,7 +35,9 @@ public class GetByIdTest extends EmbeddedSolrServerTestBase {
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    initCore();
+    solrClientTestRule.startSolr(LuceneTestCase.createTempDir("solrhome"));
```

Comment:
Why do some use `solrClientTestRule.startSolr(Paths.get(SolrTestCaseJ4.TEST_HOME()));` and some need their own temp dir like in this?   Maybe this is really asking about some documentation somewhere on how to write good tests that use this Rule?

Reply:
Usually it's a reflection of whoever wrote the test and was maybe arbitrary.  I'm not sure we can strongly advocate one or the other.  I like the empty home dir insofar as the node configuration (usually read from solr.xml) is non-existent so  then the test infrastructure is in a position to tweak things for tests; perhaps being more minimalistic.  Perhaps a no-arg startSolr() would make this easy-path even easier and thus used more, which is a good thing.

Reply:
Maybe it's just my love for the Rails world's view of "convention over configuration"...    I'd love to see us strongly advocate for one approach whenever there are two approaches available because of the wide variety of folks working on Solr.  I suspect most folks are copy/pasting to decide how to do things, and aren't thinking carefully thorugh one option or the other.   If you provide two options that appear similarily good then you will live with two ways of doing it henceforth!   I'd love to see that `startSolr()` ;-)